src/activities/Elsa.Activities.MassTransit/Activities/ReceiveMassTransitMessage.cs

```diff
@@ -32,10 +32,14 @@ public Type MessageType
 
         protected override bool OnCanExecute(WorkflowExecutionContext context)
         {
-            var message = context.Workflow.Input["message"];
+            var message = context.Workflow.Input[Constants.MessageInputKey];
             var messageType = MessageType;
 
-            return message != null && messageType != null && message.GetType() == messageType;
+            // TODO: Workout what needs to be done here as the message input is now a JObject,

```

Comment:
Good catch. I'm thinking we might just update `WorkflowConsumer<T>` to include the assembly-qualified message type name as part of the workflow input in addition to the message itself, like so:

```csharp
input.SetVariable("MessageType", typeof(T).AssemblyQualifiedName);
```

Then the `OnCanExecute` method can read the message type from the workflow input and compare that.

Reply:
Yeah sounds good. I'll hand over the type in the input from the consumer.